src/server/worker/pipeline/transform/worker.go

```diff
@@ -114,21 +115,25 @@ func handleDatumSet(driver driver.Driver, logger logs.TaggedLogger, datumSet *Da
 						}
 						if driver.PipelineInfo().Details.Transform.ErrCmd != nil {
 							opts = append(opts, datum.WithRecoveryCallback(func(runCtx context.Context) error {
-								return driver.RunUserErrorHandlingCode(runCtx, logger, env)
+								return errors.EnsureStack(driver.RunUserErrorHandlingCode(runCtx, logger, env))
 							}))
 						}
 						return s.WithDatum(meta, func(d *datum.Datum) error {
 							cancelCtx, cancel := context.WithCancel(ctx)
 							defer cancel()
-							return status.withDatum(inputs, cancel, func() error {
-								return driver.WithActiveData(inputs, d.PFSStorageRoot(), func() error {
-									return d.Run(cancelCtx, func(runCtx context.Context) error {
-										return driver.RunUserCode(runCtx, logger, env)
+							err := status.withDatum(inputs, cancel, func() error {
```

Comment:
ah, this is kind of unfortunate, but I guess wrapping everything in the function call is worse